python/dgl/data/utils.py

```diff
@@ -350,3 +361,395 @@ def __len__(self):
+    mx = r_mat_inv.dot(mx)
+    return np.array(mx.todense())
+
+def col_normalize(features):
+    mx = sp.csr_matrix(features, dtype=np.float32)
+
+    colsum = np.array(mx.sum(0))
+    c_inv = np.power(colsum, -1).flatten()
+    c_inv[np.isinf(c_inv)] = 0.
+    c_mat_inv = sp.diags(c_inv).transpose()
+    mx = mx.dot(c_mat_inv)
+    return np.array(mx.todense())
+
+def float_row_l1_normalize(features):
+    rowsum = np.sum(np.abs(features), axis=1)
+    r_inv = np.power(rowsum, -1).reshape(-1,1)
+    r_inv[np.isinf(r_inv)] = 0.
+    return features * r_inv
+
+def float_col_l1_normalize(features):
+    colsum = np.sum(np.abs(features), axis=0)
+    c_inv = np.power(colsum, -1)
+    c_inv[np.isinf(c_inv)] = 0.
+    return features * c_inv
+
+def embed_word2vec(str_val, nlps):
+    """ Use NLP encoder to encode the string into vector
+
+    There can be multiple NLP encoders in nlps. Each encoder
+    is invoded to generate a embedding for the input string and
+    the resulting embeddings are concatenated.
+
+    Parameters
+    ----------
+    str_val : str
+        words to encode
+
+    nlps : list of func
+        a list of nlp encoder functions
+    """
+    vector = None
+    for nlp in nlps:
+        doc = nlp(str_val)
+        if vector is None:
+            vector = doc.vector
+        else:
+            vector = np.concatenate((vector, doc.vector))
+    return vector
+
+def parse_lang_feat(str_feats, nlps, verbose=False):
+    """ Parse a list of strings using word2vec encoding using NLP encoders in nlps
+
+    Parameters
+    ----------
+    str_feats : list of str
+        list of strings to encode
+
+    nlps : list of func
+        a list of nlp encoder functions
+
+    verbose : bool, optional
+        print out debug info
+        Default: False
+
+    Return
+    ------
+    numpy.array
+        the encoded features
+    """
+    features = []
+    num_feats = len(str_feats)
+    num_process = num_feats if num_feats < 8 else 8 # TODO(xiangsx) get system nproc
+    batch_size = (num_feats + num_process - 1) // num_process
+
+    def embed_lang(d, proc_idx, feats):
+        res_feats = []
+        for s_feat in feats:
+            res_feats.append(embed_word2vec(s_feat, nlps))
+        d[proc_idx] = res_feats
+
+    # use multi process to process the feature
+    manager = Manager()
+    d = manager.dict()
+    job=[]
+    for i in range(num_process):
+        sub_info = str_feats[i * batch_size : (i+1) * batch_size \
+                         if (i+1) * batch_size < num_feats else num_feats]
+        job.append(Process(target=embed_lang, args=(d, i, sub_info)))
+
+    for p in job:
+        p.start()
+
+    for p in job:
+        p.join()
+
+    for i in range(num_process):
+        if len(d[i]) > 0:
+            features.append(d[i])
+
+    features = np.concatenate(features)
+    if verbose:
+        print(features.shape)
+
+    return features
+
+def parse_word2vec_node_feature(str_feats, langs, verbose=False):
```

Comment:
Why only node features?

Reply:
The name is confusing, it should be parse_word2vec_feature.